pandas/io/formats/format.py

```diff
@@ -1047,7 +1047,8 @@ def get_result_as_array(self):
         """
 
         if self.formatter is not None:
-            return np.array([self.formatter(x) for x in self.values])
+            return np.array([self.formatter(x) if not isna(x) else self.na_rep
```

Comment:
construct the array (as object), the set the nan's via mask

Reply:
I've changed that list comp. to a np.array then mask. Is that what you mean?